third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -141,37 +141,24 @@ struct ConvertLayoutOpConversion
     RankedTensorType srcTy = op.getSrc().getType();
     RankedTensorType dstTy = op.getType();
     if (isMfmaToDotShortcut(srcTy, dstTy)) {
-      // get source values
-      auto vals = unpackLLElements(loc, adaptor.getSrc(), rewriter);
-      unsigned elems = getTotalElemsPerThread(srcTy);
-      Type elemTy =
-          this->getTypeConverter()->convertType(srcTy.getElementType());
-      // for the destination type, we need to pack values together
-      // so they can be consumed by tensor core operations
-      SmallVector<Value> vecVals;
-      SmallVector<Type> types;
-      auto elemSize = elemTy.getIntOrFloatBitWidth();
-      // TODO: Support types other than float16 and
-      // bf16 (represented as int16 in llvm ir).
-      assert((type::isFloat(elemTy) || type::isInt(elemTy)) && elemSize == 16);
       // vecSize is an number of sequential elements stored by one thread
-      // - For MFMA (nonKDim == 32) encoding it is 4
-      // - For MFMA (nonKDim == 32) operand encoding it is
-      // dotOperandEndocing::kWidth,
-      //   which is 4 for fp16 and bfloat16 dtypes
+      // - For MFMA encoding (encoding of the result tensor of dot
+      // operation) it is 4
```

Comment:
I think for current MI GPUs, it is 4. But it's better if we don't hard code it and instead use something like `getContigPerThread()`

Reply:
I think for this comment, it's fine. The hard coded value "4" is in `isMfmaToDotShortcut()`. It would be better if you can fix it there.

Reply:
done